### PR TITLE
Set a better defining CWE ID to XFO

### PR DESCRIPTION
**Pull request by ThunderSon**
Replaced the CWE ID 16 with 1021, and updated the comment.
Fixes https://github.com/zaproxy/zaproxy/issues/6532

Signed-off-by: Elie Saad <eliesaad7@gmail.com>

### CODE REVIEW

File: addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java [hunk 1]
```diff
@@ -129,7 +129,8 @@ private AlertBuilder buildAlert(String evidence, VulnType currentVT) {
                 .setSolution(getAlertElement(currentVT, "soln"))
                 .setReference(getAlertElement(currentVT, "refs"))
                 .setEvidence(evidence)
-                .setCweId(16) // CWE-16: Configuration
+                .setCweId(1021) // CWE-1021: Improper Restriction of Rendered UI Layers or Frames
+                .setCweId(1021) // CWE-1021: Improper Restriction of Rendered UI Layers or Frames
```
Comment (thc202) [inline addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XFrameOptionScanRule.java:133]: The statement is duplicated.

**Comment by thc202** [comment]
The changelog should be updated.

**Comment by ThunderSon** [comment]
Force pushed again to add the commit message. Let me know if you require anything additional since it's my first PR.

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**